File: src/main/java/org/nwapw/abacus/tree/VariableNode.java
package org.nwapw.abacus.tree;

import org.nwapw.abacus.number.NumberInterface;

public class VariableNode extends TreeNode{
    private String variable;
    public VariableNode() {
        variable = null;
    }
    public VariableNode(String name){
        this.variable = name;
    }
    public String getVariable() {
        return variable;
    }
    @Override
    public <T> T reduce(Reducer<T> reducer) {
        return reducer.reduceNode(this);
    }
    @Override
    public String toString() {
        return variable;
    }
}

File: core/src/main/kotlin/org/nwapw/abacus/tree/nodes/VariableNode.kt
package org.nwapw.abacus.tree.nodes

import org.nwapw.abacus.tree.Reducer

/**
 * A tree node that holds a placeholder variable.
 *
 * This node holds a variable string, and acts similarly to a number,
 * with the key difference of not actually holding a value at runtime.
 *
 * @param variable the actual variable name that this node represents.
 */
class VariableNode(val variable: String) : TreeNode() {

    override fun <T : Any> reduce(reducer: Reducer<T>): T {
        return reducer.reduceNode(this)
    }

    override fun toString(): String {
        return variable
    }

}